test/features/profile/view/profile_view_test.dart

```diff
@@ -217,4 +215,97 @@ void main() {
+      'Given user tap on Delete Account option and confirmation dialog is showing'
+      'When user select Cancel'
+      'Then the app will dismiss the dialog', (widgetTester) async {
+    when(profileCubit.stream)
+        .thenAnswer((_) => Stream.value(const ProfileInitialState()));
+    when(profileCubit.state).thenAnswer((_) => const ProfileInitialState());
+
+    await widgetTester.blocWrapAndPump<ProfileCubit>(
+        profileCubit, const ProfileBody());
+
+    expect(find.byType(ProfileBody), findsOneWidget);
+
+    final Finder deleteAccountOptionFinder = find.text('Delete account');
+    expect(deleteAccountOptionFinder, findsOneWidget);
+
+    await widgetTester.ensureVisible(deleteAccountOptionFinder);
+    await widgetTester.tap(deleteAccountOptionFinder);
+    await widgetTester.pumpAndSettle();
+
+    expect(
+        find.ancestor(
+            of: find.text('Delete account?'),
+            matching: find.byType(HsWarningBottomSheetView)),
+        findsOneWidget);
+    final Finder cancelBtn = find.descendant(
+        of: find.byType(HsWarningBottomSheetView),
+        matching: find.text('Cancel'));
+    expect(cancelBtn, findsOneWidget);
+    await widgetTester.tap(cancelBtn);
+    await widgetTester.pumpAndSettle();
+    expect(
+        find.ancestor(
+            of: find.text('Delete account?'),
+            matching: find.byType(HsWarningBottomSheetView)),
+        findsNothing);
+  });
+
+  testWidgets(
+      'Given user tap on Delete Account option and confirmation dialog is showing'
+      'When user select Submit'
+      'Then the delete account function was called 1', (widgetTester) async {
+    when(profileCubit.stream)
+        .thenAnswer((_) => Stream.value(const ProfileInitialState()));
+    when(profileCubit.state).thenAnswer((_) => const ProfileInitialState());
+
+    await widgetTester.blocWrapAndPump<ProfileCubit>(
+        profileCubit, const ProfileBody());
+
+    expect(find.byType(ProfileBody), findsOneWidget);
+
+    final Finder deleteAccountOptionFinder = find.text('Delete account');
+    expect(deleteAccountOptionFinder, findsOneWidget);
+
+    await widgetTester.ensureVisible(deleteAccountOptionFinder);
+    await widgetTester.tap(deleteAccountOptionFinder);
+    await widgetTester.pumpAndSettle();
+
+    expect(
+        find.ancestor(
+            of: find.text('Delete account?'),
+            matching: find.byType(HsWarningBottomSheetView)),
+        findsOneWidget);
+    final Finder submitBtn = find.descendant(
+        of: find.byType(HsWarningBottomSheetView),
+        matching: find.text('Submit'));
+    expect(submitBtn, findsOneWidget);
+    await widgetTester.ensureVisible(submitBtn);
+    await widgetTester.tap(submitBtn);
+    verify(profileCubit.deleteAccount()).called(1);
+  });
+
+  testWidgets(
+      'Given delete account feature works as expected'
+      'When user delete their account'
+      'Then the app will navigate user to Login Screen', (widgetTester) async {
```

Comment:
Using signup screen for consistency with your code
```suggestion
      'Then the app will navigate user to SignUp Screen', (widgetTester) async {
```